### PR TITLE
Normalize reflection probe content with Probe Volume data

### PR DESCRIPTION
**Pull request by FrancescoC-unity**
Added option (in frame settings for now) to have reflection probe content normalized by the APV data. 

This allows for having more coherent reflections with the environment, even with a lower density of reflection probes. 

ezgif com-gif-maker

ezgif-2-1560a8486f93

The change will always bake additional SH9 per reflection probe, and the luminance of said coefficients are sent over to perform the normalization. 

You can see in gif what was tested, I would add a test but we don't have one with probe volumes being active yet (due to the experimental status of it) 

Note that this means that if probe volume data is bad so would the normalization.

### CODE REVIEW

**Comment by FrancescoC-unity** [comment]
Again formatting here is due to my hooks being messed up. I am going to fixup manually.

**Review by CertifiedRice — APPROVED** [review]

File: com.unity.render-pipelines.core/Runtime/Lighting/ProbeVolume/ProbeVolume.hlsl [hunk 1]
```diff
@@ -145,102 +163,218 @@ bool TryToGetPoolUVW(APVResources apvRes, float3 posWS, float3 normalWS, float3
...
+void EvaluateAdaptiveProbeVolume(APVSample apvSample, float3 normalWS, float3 backNormalWS, out float3 bakeDiffuseLighting, out float3 backBakeDiffuseLighting)
+{
+    if (apvSample.status != APV_SAMPLE_STATUS_INVALID)
     {
-        float unused;
-        float3 L0 = EvaluateAPVL0(apvRes, pool_uvw, unused);
-        bakeDiffuseLighting = L0;
+        apvSample.Decode();
+
+#ifdef PROBE_VOLUMES_L1
```
[the remark this answers is not in the document]
Comment (sebastienlagarde) [reply com.unity.render-pipelines.core/Runtime/Lighting/ProbeVolume/ProbeVolume.hlsl:247]: isapvSample.L0; fill?

[the remark this answers is not in the document]
Comment (FrancescoC-unity) [reply com.unity.render-pipelines.core/Runtime/Lighting/ProbeVolume/ProbeVolume.hlsl:247]: If we reach this file there is a guarantee that we have one of them defined, if none of them are defined we have bigger problems at caller level

[the remark this answers is not in the document]
Comment (FrancescoC-unity) [reply com.unity.render-pipelines.core/Runtime/Lighting/ProbeVolume/ProbeVolume.hlsl:247]: Yeah the L0 is always filled upon the SampleAPV call

File: com.unity.render-pipelines.core/Runtime/Lighting/ProbeVolume/ProbeVolume.hlsl [hunk 2]
```diff
@@ -145,102 +163,218 @@ bool TryToGetPoolUVW(APVResources apvRes, float3 posWS, float3 normalWS, float3
...
+    }
+}
+
+// -------------------------------------------------------------
+// Reflection Probe Normalization functions
+// -------------------------------------------------------------
+
+
+// Same idea as in Rendering of COD:IW [Drobot 2017]
+float GetReflProbeNormalizationFactor(float3 sampleDir, float4 reflProbeSHL0L1, float4 reflProbeSHL2_1, float reflProbeSHL2_2)
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.core/Runtime/Lighting/ProbeVolume/ProbeVolume.hlsl:348]: Regarding naming, it is a bit weird
GetReflProbeNormalizationFactor
GetReflectionProbeNormalizationFactor
can't you both call them "GetReflectionProbeNormalizationFactor" ? they have different attribute, 
or maybe one Internal_GetReflectionProbeNormalizationFactor() ?

Comment (FrancescoC-unity) [reply com.unity.render-pipelines.core/Runtime/Lighting/ProbeVolume/ProbeVolume.hlsl:348]: Definitively yes, will rename

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.hlsl [hunk 3]
```diff
@@ -401,7 +469,20 @@ void LightLoop( float3 V, PositionInputs posInput, PreLightData preLightData, BS
                     v_envLightListOffset++;
                     if (reflectionHierarchyWeight < 1.0)
                     {
-                        EVALUATE_BSDF_ENV(s_envLightData, REFLECTION, reflection);
+                        if (IsMatchingLightLayer(s_envLightData.lightLayers, builtinData.renderingLayers))
+                        {
+                            IndirectLighting lighting = EvaluateBSDF_Env(context, V, posInput, preLightData, s_envLightData, bsdfData, s_envLightData.influenceShapeType, GPUIMAGEBASEDLIGHTINGTYPE_REFLECTION, reflectionHierarchyWeight);
+#if defined(PROBE_VOLUMES_L1) || defined(PROBE_VOLUMES_L2)
+
+                            if (s_envLightData.normalizeWithAPV > 0 && all(lightInReflDir >= 0))
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.hlsl:477]: this part miss the #if defined(PROBE_VOLUMES_L1) || defined(PROBE_VOLUMES_L

**Review by sebastienlagarde — APPROVED** [review]

**Pull request merged**